.env/src/cb.py

```diff
@@ -0,0 +1,222 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from enum import Enum, auto
+from typing import List
+
+from util import F
+from expr import Expr, Const, Neg, to_expr, ToExpr
+from query import StepTypeNext
+from chiquito_ast import StepType
+
+##########
+# dsl/cb #
+##########
+
+
+class Typing(Enum):
+    Unknown = auto()
+    Boolean = auto()
+    AntiBooly = auto()
+
+
+@dataclass
+class Constraint:
+    annotation: str
+    expr: Expr
+    typing: Typing
+
+    def from_expr(
+        expr: Expr,
+    ) -> Constraint:  # Cannot call function `from`, a reserved keyword in Python.
+        annotation: str = str(expr)
+        if isinstance(expr, StepTypeNext):
+            return Constraint(annotation, expr, Typing.Boolean)
+        else:
+            return Constraint(annotation, expr, Typing.Unknown)
+
+    def __str__(self: Constraint) -> str:
+        return self.annotation
+
+    def cb_and(
```

Comment:
@qwang98 will this mean we need to call Constraint.cb_and(....) ? Can we overload the and operator? Or is there a way to make it less verbose?

Reply:
Is it necessary that this functions have to be members of the Constraint class? Cannot be simple functions?

Reply:
I don't think it's possible to overload `and` `or` `not` in Python. You can overload `&` `|` with `__and__` `__or__`, but can't overload the literal `and` `or`.

They can be simple functions and I will change all of them to behave like that!

Reply:
Fixed! Made all cb functions simple functions so we don't need to Constraint.eq, for example, any more.